src/control/IntersectionReferenceSystem.ts

```diff
@@ -258,8 +278,16 @@ export class IntersectionReferenceSystem {
 
     const totalLength = this.displacement + extensionStart + extensionEnd;
     const preSteps = Math.floor((extensionStart / totalLength) * steps);
-    const curveSteps = Math.max(Math.ceil((this.displacement / totalLength) * steps), 1);
-    const postSteps = steps - curveSteps - preSteps;
+    let curveSteps = Math.max(Math.ceil((this.displacement / totalLength) * steps), 1);
+
+    // The curve interpolator returns steps + 1 number of points
+    // Subtract the extra point from curveSteps if enough points
+    const extraCurveStep = 1;
+    if (curveSteps > 1) {
+      curveSteps -= extraCurveStep;
+    }
+
+    const postSteps = steps - (curveSteps + extraCurveStep) - preSteps;
 
```

Comment:
This seems like a bit of extra code rather than starting the iteration at 1 or does it fix something else?

Reply:
There are two different issues tackled here.

1. In existing code the loop starts at 1, so the points generated are postSteps - 1. The total length of the end extension will not be reached when the last point missing. We could change the calculation `const f = (i + 1) * postStep;` to `const f = (i + 1) * (postStep-1);` but not sure if that is any better.

2. Since the curve interpolator returns curveSteps + 1 we have to account for that by reducing the number of steps somewhere. In existing code it's removed from the postSteps without documenting or making it obvious. That might be a ok solution if we document it, but the prePoints and postPoints will differ by up to 2 points. Is that an issue, or can we live with that?

Reply:
To get closer to a correct distribution of the points I try to remove the extra point from curveSteps if possible.
This is not a pretty solution so I could have another go at it to make it simpler if you'd like. Alternatively if the distribution of points isn't all that important down to the last 1-2 points, I could remove this fix, and keeps it as it was.

Reply:
I must admit I was puzzled when I saw the interpolator returned +1 to what is passed in, but thinking of this as steps or splits rather than points it makes sense ie. if we want to split the path into two "steps", we need three points (endpoints and middle). I'm not sure if this is consistent through the code neither with naming nor behavior (probably not). Iterations of post steps start at 1 since the last point returned by the interpolator is the starting point. I would say the correct behavior would be that the distance between points is as uniform as possible and the extension lengths are correct, but it's going to work in rep regardless. I think this could do with a good cleanup, even thought there will be a minor issue is when the trajectory calculations change all the seismic slices will need to be clear from cache from the seismic API and recalculated as it's cached on a hash calculated from the trajectory. But maybe the cleanup should be a separate task

Reply:
I have removed this part from the PR as this does not need to be solved in this PR anymore, and created an issue for it here: https://github.com/equinor/esv-intersection/issues/383
This should be looked at since the endExtension is shorter than it should, and even though we are talking about steps and not points it seems to me like start and end extensions are not getting an equal amount of steps.